### PR TITLE
Fix index out of range issue when no args are provided.

### LINKED ISSUE #404: Error when running ./mgc
I get the following when running the Microsoft Graph CLI 1.1.0 GA ./mgc in a terminal (zsh) on a Macbook Pro M3:

Unhandled exception. System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Microsoft.Graph.Cli.Program.Main(String[] args)
   at Microsoft.Graph.Cli.Program.<Main>(String[] args)
zsh: abort      ./mgc

### PR DESCRIPTION
**Pull request by calebkiage**
Closes #404

### CODE REVIEW

File: src/Program.cs [hunk 1]
```diff
@@ -47,7 +47,7 @@ static async Task<int> Main(string[] args)
             Console.InputEncoding = Encoding.UTF8;
             Console.OutputEncoding = Encoding.UTF8;
             // Replace `me ...` with `users ... --user-id me`
-            if (args[0] == "me")
+            if (args.Length > 1 && args[0] == "me")
```
Comment (baywet) [inline src/Program.cs]:
```suggestion
            if (args.Length > 0 && args[0] == "me")
```

File: .azure-pipelines/release-cli.yaml [hunk 2]
```diff
@@ -411,7 +411,7 @@ stages:
           # Nuget tool doesn't work with multi-stage builds
           - task: UseDotNet@2
             displayName: 'Use .NET 7'
-            condition: and(succeeded(), ne('${{ parameters.simulate }}', 'true'), eq(variables['IS_NUGET'], 'true'))
+            condition: and(false, succeeded(), ne('${{ parameters.simulate }}', 'true'), eq(variables['IS_NUGET'], 'true'))
```
Comment (baywet) [inline .azure-pipelines/release-cli.yaml:414]: I'm not sure why we're disabling all those tasks?

**Review by baywet — CHANGES_REQUESTED** [review]

File: .azure-pipelines/release-cli.yaml (hunk 2 above)
Comment (calebkiage) [reply .azure-pipelines/release-cli.yaml:414]: We don't publish to NuGet anymore

File: src/Program.cs (hunk 1 above)
Comment (calebkiage) [reply src/Program.cs]: Writing C++ has confused me a little 😅

File: .azure-pipelines/release-cli.yaml (hunk 2 above)
Comment (baywet) [reply .azure-pipelines/release-cli.yaml:414]: right, I had forgotten

**Review by baywet — APPROVED** [review]

**Pull request merged**